internal/cluster/distribution/eks/eksprovider/workflow/eks.go

```diff
@@ -0,0 +1,61 @@
+// Copyright © 2020 Banzai Cloud
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package workflow
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/client"
+	"github.com/aws/aws-sdk-go/service/eks"
+	eksi "github.com/aws/aws-sdk-go/service/eks/eksiface"
+)
+
+// +testify:mock
+
+// eksAPI redefines the eksiface.EKSAPI
+// interface in order to generate mock for it.
+// nolint:deadcode // Used for mock generation and only the original interface
+// is referenced.
+type eksAPI interface {
+	eksi.EKSAPI
+}
+
+// +testify:mock
+
+// EKSAPIFactory provides an interface for instantiating AWS
+// EKS API objects.
+type EKSAPIFactory interface {
```

Comment:
This is just a wrapper to not to rely directly on the AWS lib, for mocking. @pregnor am I right?

Reply:
That's correct, we use it for test mocking purposes to be able to provide a mocked API in place of an `eksiface.EKSAPI`, we did it similarly with the `CloudFormation` factories and client/service interfaces.

Context: It helps the Cadence activities to have a constructor argument/field for factories returning client/service interfaces to the corresponding AWS APIs so we don't use `eks.New(awsSession)` because it makes test mocking much more difficult as it creates a real API object, this way we can use a `EKSFactoryMock{}` object in place of the real factory wrapping the `New()` call which allows us to write extensive mock tests.